From 026f8041a248d2a556b39ea4463068d454b05944 Mon Sep 17 00:00:00 2001
From: Nadav Haklai <nadavh@marvell.com>
Date: Tue, 29 Mar 2016 17:27:22 +0300
Subject: [PATCH 088/239] mv_ddr: Relocate res ctrl byte/bit regs defs per plat

This patch relocates DDR subphys' result control per byte/bit
registers to be per platform. (These registers are differnt
between the platforms.)

Change-Id: Ic1789c1b4855a6558d257d27d423fe90f237f62b
Signed-off-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28640
Reviewed-on: http://vgitil04.il.marvell.com:8080/28795
Tested-by: Star_Automation <star@marvell.com>
---
 drivers/marvell/mv_ddr/a38x/mv_ddr_a38x.h      | 53 +++++++++++++++
 drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.h  | 93 ++++++++++++++++++++++++++
 drivers/marvell/mv_ddr/ddr3_training_ip_flow.h | 47 -------------
 3 files changed, 146 insertions(+), 47 deletions(-)

diff --git a/drivers/marvell/mv_ddr/a38x/mv_ddr_a38x.h b/drivers/marvell/mv_ddr/a38x/mv_ddr_a38x.h
index 21d03a8..c32270b 100644
--- a/drivers/marvell/mv_ddr/a38x/mv_ddr_a38x.h
+++ b/drivers/marvell/mv_ddr/a38x/mv_ddr_a38x.h
@@ -182,6 +182,59 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #define MAX_DQ_NUM				40
 
+/* Subphy result control per byte registers */
+#define RESULT_CONTROL_BYTE_PUP_0_REG		0x1830
+#define RESULT_CONTROL_BYTE_PUP_1_REG		0x1834
+#define RESULT_CONTROL_BYTE_PUP_2_REG		0x1838
+#define RESULT_CONTROL_BYTE_PUP_3_REG		0x183c
+#define RESULT_CONTROL_BYTE_PUP_4_REG		0x18b0
+
+/* Subphy result control per bit registers */
+#define RESULT_CONTROL_PUP_0_BIT_0_REG		0x18b4
+#define RESULT_CONTROL_PUP_0_BIT_1_REG		0x18b8
+#define RESULT_CONTROL_PUP_0_BIT_2_REG		0x18bc
+#define RESULT_CONTROL_PUP_0_BIT_3_REG		0x18c0
+#define RESULT_CONTROL_PUP_0_BIT_4_REG		0x18c4
+#define RESULT_CONTROL_PUP_0_BIT_5_REG		0x18c8
+#define RESULT_CONTROL_PUP_0_BIT_6_REG		0x18cc
+#define RESULT_CONTROL_PUP_0_BIT_7_REG		0x18f0
+
+#define RESULT_CONTROL_PUP_1_BIT_0_REG		0x18f4
+#define RESULT_CONTROL_PUP_1_BIT_1_REG		0x18f8
+#define RESULT_CONTROL_PUP_1_BIT_2_REG		0x18fc
+#define RESULT_CONTROL_PUP_1_BIT_3_REG		0x1930
+#define RESULT_CONTROL_PUP_1_BIT_4_REG		0x1934
+#define RESULT_CONTROL_PUP_1_BIT_5_REG		0x1938
+#define RESULT_CONTROL_PUP_1_BIT_6_REG		0x193c
+#define RESULT_CONTROL_PUP_1_BIT_7_REG		0x19b0
+
+#define RESULT_CONTROL_PUP_2_BIT_0_REG		0x19b4
+#define RESULT_CONTROL_PUP_2_BIT_1_REG		0x19b8
+#define RESULT_CONTROL_PUP_2_BIT_2_REG		0x19bc
+#define RESULT_CONTROL_PUP_2_BIT_3_REG		0x19c0
+#define RESULT_CONTROL_PUP_2_BIT_4_REG		0x19c4
+#define RESULT_CONTROL_PUP_2_BIT_5_REG		0x19c8
+#define RESULT_CONTROL_PUP_2_BIT_6_REG		0x19cc
+#define RESULT_CONTROL_PUP_2_BIT_7_REG		0x19f0
+
+#define RESULT_CONTROL_PUP_3_BIT_0_REG		0x19f4
+#define RESULT_CONTROL_PUP_3_BIT_1_REG		0x19f8
+#define RESULT_CONTROL_PUP_3_BIT_2_REG		0x19fc
+#define RESULT_CONTROL_PUP_3_BIT_3_REG		0x1a30
+#define RESULT_CONTROL_PUP_3_BIT_4_REG		0x1a34
+#define RESULT_CONTROL_PUP_3_BIT_5_REG		0x1a38
+#define RESULT_CONTROL_PUP_3_BIT_6_REG		0x1a3c
+#define RESULT_CONTROL_PUP_3_BIT_7_REG		0x1ab0
+
+#define RESULT_CONTROL_PUP_4_BIT_0_REG		0x1ab4
+#define RESULT_CONTROL_PUP_4_BIT_1_REG		0x1ab8
+#define RESULT_CONTROL_PUP_4_BIT_2_REG		0x1abc
+#define RESULT_CONTROL_PUP_4_BIT_3_REG		0x1ac0
+#define RESULT_CONTROL_PUP_4_BIT_4_REG		0x1ac4
+#define RESULT_CONTROL_PUP_4_BIT_5_REG		0x1ac8
+#define RESULT_CONTROL_PUP_4_BIT_6_REG		0x1acc
+#define RESULT_CONTROL_PUP_4_BIT_7_REG		0x1af0
+
 /* Matrix enables DRAM modes (bus width/ECC) per boardId */
 #define TOPOLOGY_UPDATE_32BIT			0
 #define TOPOLOGY_UPDATE_32BIT_ECC		1
diff --git a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.h b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.h
index 8145b80..431d1fe 100644
--- a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.h
+++ b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.h
@@ -216,6 +216,99 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #define MAX_DQ_NUM				72
 
+/* Subphy result control per byte registers */
+#define RESULT_CONTROL_BYTE_PUP_0_REG	0x1830
+#define RESULT_CONTROL_BYTE_PUP_1_REG	0x1834
+#define RESULT_CONTROL_BYTE_PUP_2_REG	0x1838
+#define RESULT_CONTROL_BYTE_PUP_3_REG	0x183C
+#define RESULT_CONTROL_BYTE_PUP_4_REG	0x18B0
+#define RESULT_CONTROL_BYTE_PUP_5_REG	0x18B4
+#define RESULT_CONTROL_BYTE_PUP_6_REG	0x18B8
+#define RESULT_CONTROL_BYTE_PUP_7_REG	0x18BC
+#define RESULT_CONTROL_BYTE_PUP_8_REG	0x18C0
+
+/* Subphy result control per bit registers */
+#define RESULT_CONTROL_PUP_0_BIT_0_REG	0x18C4
+#define RESULT_CONTROL_PUP_0_BIT_1_REG	0x18C8
+#define RESULT_CONTROL_PUP_0_BIT_2_REG	0x18CC
+#define RESULT_CONTROL_PUP_0_BIT_3_REG	0x18F0
+#define RESULT_CONTROL_PUP_0_BIT_4_REG	0x18F4
+#define RESULT_CONTROL_PUP_0_BIT_5_REG	0x18F8
+#define RESULT_CONTROL_PUP_0_BIT_6_REG	0x18FC
+#define RESULT_CONTROL_PUP_0_BIT_7_REG	0x1930
+
+#define RESULT_CONTROL_PUP_1_BIT_0_REG	0x1934
+#define RESULT_CONTROL_PUP_1_BIT_1_REG	0x1938
+#define RESULT_CONTROL_PUP_1_BIT_2_REG	0x193C
+#define RESULT_CONTROL_PUP_1_BIT_3_REG	0x19B0
+#define RESULT_CONTROL_PUP_1_BIT_4_REG	0x19B4
+#define RESULT_CONTROL_PUP_1_BIT_5_REG	0x19B8
+#define RESULT_CONTROL_PUP_1_BIT_6_REG	0x19BC
+#define RESULT_CONTROL_PUP_1_BIT_7_REG	0x19C0
+
+#define RESULT_CONTROL_PUP_2_BIT_0_REG	0x19C4
+#define RESULT_CONTROL_PUP_2_BIT_1_REG	0x19C8
+#define RESULT_CONTROL_PUP_2_BIT_2_REG	0x19CC
+#define RESULT_CONTROL_PUP_2_BIT_3_REG	0x19F0
+#define RESULT_CONTROL_PUP_2_BIT_4_REG	0x19F4
+#define RESULT_CONTROL_PUP_2_BIT_5_REG	0x19F8
+#define RESULT_CONTROL_PUP_2_BIT_6_REG	0x19FC
+#define RESULT_CONTROL_PUP_2_BIT_7_REG	0x1A30
+
+#define RESULT_CONTROL_PUP_3_BIT_0_REG	0x1A34
+#define RESULT_CONTROL_PUP_3_BIT_1_REG	0x1A38
+#define RESULT_CONTROL_PUP_3_BIT_2_REG	0x1A3C
+#define RESULT_CONTROL_PUP_3_BIT_3_REG	0x1AB0
+#define RESULT_CONTROL_PUP_3_BIT_4_REG	0x1AB4
+#define RESULT_CONTROL_PUP_3_BIT_5_REG	0x1AB8
+#define RESULT_CONTROL_PUP_3_BIT_6_REG	0x1ABC
+#define RESULT_CONTROL_PUP_3_BIT_7_REG	0x1AC0
+
+#define RESULT_CONTROL_PUP_4_BIT_0_REG	0x1AC4
+#define RESULT_CONTROL_PUP_4_BIT_1_REG	0x1AC8
+#define RESULT_CONTROL_PUP_4_BIT_2_REG	0x1ACC
+#define RESULT_CONTROL_PUP_4_BIT_3_REG	0x1AF0
+#define RESULT_CONTROL_PUP_4_BIT_4_REG	0x1AF4
+#define RESULT_CONTROL_PUP_4_BIT_5_REG	0x1AF8
+#define RESULT_CONTROL_PUP_4_BIT_6_REG	0x1AFC
+#define RESULT_CONTROL_PUP_4_BIT_7_REG	0x1B30
+
+#define RESULT_CONTROL_PUP_5_BIT_0_REG	0x1B34
+#define RESULT_CONTROL_PUP_5_BIT_1_REG	0x1B38
+#define RESULT_CONTROL_PUP_5_BIT_2_REG	0x1B3C
+#define RESULT_CONTROL_PUP_5_BIT_3_REG	0x1BB0
+#define RESULT_CONTROL_PUP_5_BIT_4_REG	0x1BB4
+#define RESULT_CONTROL_PUP_5_BIT_5_REG	0x1BB8
+#define RESULT_CONTROL_PUP_5_BIT_6_REG	0x1BBC
+#define RESULT_CONTROL_PUP_5_BIT_7_REG	0x1BC0
+
+#define RESULT_CONTROL_PUP_6_BIT_0_REG	0x1BC4
+#define RESULT_CONTROL_PUP_6_BIT_1_REG	0x1BC8
+#define RESULT_CONTROL_PUP_6_BIT_2_REG	0x1BCC
+#define RESULT_CONTROL_PUP_6_BIT_3_REG	0x1BF0
+#define RESULT_CONTROL_PUP_6_BIT_4_REG	0x1BF4
+#define RESULT_CONTROL_PUP_6_BIT_5_REG	0x1BF8
+#define RESULT_CONTROL_PUP_6_BIT_6_REG	0x1BFC
+#define RESULT_CONTROL_PUP_6_BIT_7_REG	0x1C30
+
+#define RESULT_CONTROL_PUP_7_BIT_0_REG	0x1C34
+#define RESULT_CONTROL_PUP_7_BIT_1_REG	0x1C38
+#define RESULT_CONTROL_PUP_7_BIT_2_REG	0x1C3C
+#define RESULT_CONTROL_PUP_7_BIT_3_REG	0x1CB0
+#define RESULT_CONTROL_PUP_7_BIT_4_REG	0x1CB4
+#define RESULT_CONTROL_PUP_7_BIT_5_REG	0x1CB8
+#define RESULT_CONTROL_PUP_7_BIT_6_REG	0x1CBC
+#define RESULT_CONTROL_PUP_7_BIT_7_REG	0x1CC0
+
+#define RESULT_CONTROL_PUP_8_BIT_0_REG	0x1CC4
+#define RESULT_CONTROL_PUP_8_BIT_1_REG	0x1CC8
+#define RESULT_CONTROL_PUP_8_BIT_2_REG	0x1CCC
+#define RESULT_CONTROL_PUP_8_BIT_3_REG	0x1CF0
+#define RESULT_CONTROL_PUP_8_BIT_4_REG	0x1CF4
+#define RESULT_CONTROL_PUP_8_BIT_5_REG	0x1CF8
+#define RESULT_CONTROL_PUP_8_BIT_6_REG	0x1CFC
+#define RESULT_CONTROL_PUP_8_BIT_7_REG	0x1D30
+
 /* Matrix enables DRAM modes (bus width/ECC) per boardId */
 #define TOPOLOGY_UPDATE_32BIT			0
 #define TOPOLOGY_UPDATE_32BIT_ECC		1
diff --git a/drivers/marvell/mv_ddr/ddr3_training_ip_flow.h b/drivers/marvell/mv_ddr/ddr3_training_ip_flow.h
index a717699..3040596 100644
--- a/drivers/marvell/mv_ddr/ddr3_training_ip_flow.h
+++ b/drivers/marvell/mv_ddr/ddr3_training_ip_flow.h
@@ -270,53 +270,6 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define ODPG_BIST_DONE_BIT_VALUE_REV2		1
 #define ODPG_BIST_DONE_BIT_VALUE_REV3		0
 
-#define RESULT_CONTROL_BYTE_PUP_0_REG		0x1830
-#define RESULT_CONTROL_BYTE_PUP_1_REG		0x1834
-#define RESULT_CONTROL_BYTE_PUP_2_REG		0x1838
-#define RESULT_CONTROL_BYTE_PUP_3_REG		0x183c
-#define RESULT_CONTROL_BYTE_PUP_4_REG		0x18b0
-
-#define RESULT_CONTROL_PUP_0_BIT_0_REG		0x18b4
-#define RESULT_CONTROL_PUP_0_BIT_1_REG		0x18b8
-#define RESULT_CONTROL_PUP_0_BIT_2_REG		0x18bc
-#define RESULT_CONTROL_PUP_0_BIT_3_REG		0x18c0
-#define RESULT_CONTROL_PUP_0_BIT_4_REG		0x18c4
-#define RESULT_CONTROL_PUP_0_BIT_5_REG		0x18c8
-#define RESULT_CONTROL_PUP_0_BIT_6_REG		0x18cc
-#define RESULT_CONTROL_PUP_0_BIT_7_REG		0x18f0
-#define RESULT_CONTROL_PUP_1_BIT_0_REG		0x18f4
-#define RESULT_CONTROL_PUP_1_BIT_1_REG		0x18f8
-#define RESULT_CONTROL_PUP_1_BIT_2_REG		0x18fc
-#define RESULT_CONTROL_PUP_1_BIT_3_REG		0x1930
-#define RESULT_CONTROL_PUP_1_BIT_4_REG		0x1934
-#define RESULT_CONTROL_PUP_1_BIT_5_REG		0x1938
-#define RESULT_CONTROL_PUP_1_BIT_6_REG		0x193c
-#define RESULT_CONTROL_PUP_1_BIT_7_REG		0x19b0
-#define RESULT_CONTROL_PUP_2_BIT_0_REG		0x19b4
-#define RESULT_CONTROL_PUP_2_BIT_1_REG		0x19b8
-#define RESULT_CONTROL_PUP_2_BIT_2_REG		0x19bc
-#define RESULT_CONTROL_PUP_2_BIT_3_REG		0x19c0
-#define RESULT_CONTROL_PUP_2_BIT_4_REG		0x19c4
-#define RESULT_CONTROL_PUP_2_BIT_5_REG		0x19c8
-#define RESULT_CONTROL_PUP_2_BIT_6_REG		0x19cc
-#define RESULT_CONTROL_PUP_2_BIT_7_REG		0x19f0
-#define RESULT_CONTROL_PUP_3_BIT_0_REG		0x19f4
-#define RESULT_CONTROL_PUP_3_BIT_1_REG		0x19f8
-#define RESULT_CONTROL_PUP_3_BIT_2_REG		0x19fc
-#define RESULT_CONTROL_PUP_3_BIT_3_REG		0x1a30
-#define RESULT_CONTROL_PUP_3_BIT_4_REG		0x1a34
-#define RESULT_CONTROL_PUP_3_BIT_5_REG		0x1a38
-#define RESULT_CONTROL_PUP_3_BIT_6_REG		0x1a3c
-#define RESULT_CONTROL_PUP_3_BIT_7_REG		0x1ab0
-#define RESULT_CONTROL_PUP_4_BIT_0_REG		0x1ab4
-#define RESULT_CONTROL_PUP_4_BIT_1_REG		0x1ab8
-#define RESULT_CONTROL_PUP_4_BIT_2_REG		0x1abc
-#define RESULT_CONTROL_PUP_4_BIT_3_REG		0x1ac0
-#define RESULT_CONTROL_PUP_4_BIT_4_REG		0x1ac4
-#define RESULT_CONTROL_PUP_4_BIT_5_REG		0x1ac8
-#define RESULT_CONTROL_PUP_4_BIT_6_REG		0x1acc
-#define RESULT_CONTROL_PUP_4_BIT_7_REG		0x1af0
-
 #define WL_PHY_REG				0x0
 #define WRITE_CENTRALIZATION_PHY_REG		0x1
 #define RL_PHY_REG				0x2
-- 
1.9.1

